python/Ice/bidir/client/clock.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) ZeroC, Inc.
+
+from EarlyRiser import AlarmClock, ButtonPressed
+import Ice
+
+class MockAlarmClock(AlarmClock):
+    """
+    MockAlarmClock is an Ice servant that implements Slice interface AlarmClock
+    """
+
+    def __init__(self, eventLoop):
+        self._needMoreTime = True
+        self._eventLoop = eventLoop
+        self._stopPressed = eventLoop.create_future()
+
+    async def waitForStopPressed(self) -> None:
+        """
+        Asynchronously wait for the ring method to return ButtonPressed.Stop.
+        """
+        await self._stopPressed
+
+    async def ring(self, message: str, current: Ice.Current) -> ButtonPressed:
```

Comment:
I am confused here, and the reader may be confused too.

Does `ring` run in the event loop thread even if it's all synchronous?